### PR TITLE
Add bank accounts information

### PR DESCRIPTION
**Pull request by PunkSage**

### CODE REVIEW

File: src/server/routes/supplier_banks.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+module.exports = function (app, db, config) {
+  SupplierBank.init(db, config).then(() =>
+  {
+    app.get('/api/suppliers/:supplierId/banks', (req, res) => sendSupplierBanks(req, res));
+    app.post('/api/suppliers/:supplierId/banks', (req, res) => createSupplierBank(req, res));
+    app.get('/api/suppliers/:supplierId/banks/:bankAccountId', (req, res) => sendSupplierBank(req, res));
+    app.put('/api/suppliers/:supplierId/banks/:bankAccountId', (req, res) => updateSupplierBank(req, res));
+    app.delete('/api/suppliers/:supplierId/banks/:bankAccountId', (req, res) => deleteSupplierBank(req, res));
+  });
```
Comment (gr4per) [inline src/server/routes/supplier_banks.js]: Do we need explicit supplierId in the url?
As this api is non public the supplierid would be implicit by user belonging to supplier

Comment (gr4per) [inline src/server/routes/supplier_banks.js]: At least authorization is missing. Currently every supplier would be able to get other suppliers bank details.
Proposed solution is to remove supplierid from url and use implicit from user data. 
By that cross tenant access is already impossible

File: src/server/db/models/SupplierBankAccount.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
+'use strict';
```
Comment (gr4per) [inline src/server/db/models/SupplierBankAccount.js]: We need syntax validation on server and client for well defined formats.
Eg Ivan and swift. For Iban also the checksum (pos 3+4,  after country code) need to be implemented...

**Comment by lasong** [comment]
LGTM

File: src/client/components/SupplierBankAccountEditor/SupplierBankAccountEditor.react.js [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+      } else if (editMode === 'view') {
+        newState.editMode = 'edit';
+      }
+      this.setState(newState);
+    }
+  }
+
+  handleDelete = (account) => {
+
+    console.log('Account', account);
```
Comment (lasong) [inline src/client/components/SupplierBankAccountEditor/SupplierBankAccountEditor.react.js]: Remove this

File: src/client/components/SupplierBankAccountEditor/SupplierBankAccountEditor.react.js [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+        accounts.splice(index, 1);
+
+        const message = this.context.i18n.getMessage('SupplierBankAccountEditor.Message.objectDeleted');
+        this.setState({ accounts: accounts, account: null, globalMessage: message, globalError: null });
+      }).catch((response) => {
+        if (response.status === 401) {
+          this.props.onUnauthorized();
+        } else {
+          console.log(`Bad request by SupplierID=${supplierId} and ContactID=${account.bankAccountId}`);
+          console.log(response);
```
Comment (lasong) [inline src/client/components/SupplierBankAccountEditor/SupplierBankAccountEditor.react.js]: Remove this

File: src/client/components/SupplierBankAccountEditor/SupplierBankAccountEditor.react.js [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+    let supplierId = this.props.supplierId;
+    account.changedBy = this.props.username;// eslint-disable-line no-param-reassign
+
+    let arg0 = encodeURIComponent(supplierId);
+    let arg1 = encodeURIComponent(account.bankAccountId);
+    request.put(`${actionUrl}/supplier/api/suppliers/${arg0}/bank_accounts/${arg1}`).
+      set('Accept', 'application/json').
+      send(account).
+      then((response) => {
+        console.log(response);
```
Comment (lasong) [inline src/client/components/SupplierBankAccountEditor/SupplierBankAccountEditor.react.js]: same as above

File: src/client/components/SupplierBankAccountEditor/SupplierBankAccountListTable.react.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  static propTypes = {
+    accounts: React.PropTypes.array.isRequired,
+    onEdit: React.PropTypes.func.isRequired,
+    onDelete: React.PropTypes.func.isRequired,
+    onView: React.PropTypes.func.isRequired,
+    readOnly: React.PropTypes.bool
+  };
+
+  onEdit = (account) => {
+    console.log(account);
```
Comment (lasong) [inline src/client/components/SupplierBankAccountEditor/SupplierBankAccountListTable.react.js]: remove

**Pull request merged**